src/kernel-install/kernel-install.c

```diff
@@ -1086,32 +1087,69 @@ static int verb_inspect(int argc, char *argv[], void *userdata) {
+
+                r = context_set_initrds(c, strv_skip(argv, 3));
+                if (r < 0)
+                        return r;
         }
 
         r = context_prepare_execution(c);
         if (r < 0)
                 return r;
 
-        printf("%sBoot Loader Entries:%s\n", ansi_underline(), ansi_normal());
-        printf("    $BOOT: %s\n", c->boot_root);
-        printf("    Token: %s\n", c->entry_token);
-        puts("");
+        if (arg_json_format_flags & JSON_FORMAT_OFF) {
+                printf("%sBoot Loader Entries:%s\n", ansi_underline(), ansi_normal());
+                printf("    $BOOT: %s\n", c->boot_root);
+                printf("    Token: %s\n", c->entry_token);
+                puts("");
 
-        printf("%sUsing plugins:%s\n", ansi_underline(), ansi_normal());
-        strv_print_full(c->plugins, "    ");
-        puts("");
+                printf("%sUsing plugins:%s\n", ansi_underline(), ansi_normal());
+                strv_print_full(c->plugins, "    ");
+                puts("");
 
-        printf("%sPlugin environment:%s\n", ansi_underline(), ansi_normal());
-        strv_print_full(c->envp, "    ");
-        puts("");
+                printf("%sPlugin environment:%s\n", ansi_underline(), ansi_normal());
+                strv_print_full(c->envp, "    ");
+                puts("");
 
-        printf("%sPlugin arguments:%s\n", ansi_underline(), ansi_normal());
-        joined = strv_join(strv_skip(c->argv, 1), " ");
-        printf("    %s\n", strna(joined));
+                printf("%sPlugin arguments:%s\n", ansi_underline(), ansi_normal());
+                joined = strv_join(strv_skip(c->argv, 1), " ");
+                printf("    %s\n", strna(joined));
+        } else {
+                _cleanup_(json_variant_unrefp) JsonVariant *v = NULL;
+
+                r = json_build(&v, JSON_BUILD_OBJECT(
+                                JSON_BUILD_PAIR_UUID("MachineId", c->machine_id),
+                                JSON_BUILD_PAIR_STRING("KernelImageType", kernel_image_type_to_string(c->kernel_image_type)),
+                                JSON_BUILD_PAIR_STRING("Layout", context_get_layout(c)),
+                                JSON_BUILD_PAIR_STRING("BootRoot", c->boot_root),
+                                JSON_BUILD_PAIR_STRING("EntryTokenType", boot_entry_token_type_to_string(c->entry_token_type)),
+                                JSON_BUILD_PAIR_STRING("EntryToken", c->entry_token),
+                                JSON_BUILD_PAIR_STRING("EntryDirectory", c->entry_dir),
+                                JSON_BUILD_PAIR_STRING("Version", c->version),
+                                JSON_BUILD_PAIR_STRING("Kernel", c->kernel),
+                                JSON_BUILD_PAIR_STRV("Initrds", c->initrds),
+                                JSON_BUILD_PAIR_STRING("InitrdGenerator", c->initrd_generator),
+                                JSON_BUILD_PAIR_STRING("UkiGenerator", c->uki_generator),
```

Comment:
What is our policy with capitalization of already capitalized words? Should this be "UKIGenerator"?